### PR TITLE
Document `snap_getClientStatus`

### LINKED ISSUE #1118: Snaps: Document `snap_getClientStatus`
Add reference documentation for the method and mention in other places (like the cron jobs section) that developers should check if the extension is locked to ensure that they don't attempt to do something that doesn't work when the extension is locked. 

https://github.com/MetaMask/snaps/blob/3f0705442d90b65c94e09b21cf2406e8e7614e70/packages/snaps-rpc-methods/src/permitted/getClientStatus.test.ts#L4

### PR DESCRIPTION
**Pull request by ziad-saab**
Closes #1118 .

### CODE REVIEW

**Review by alexandratran — APPROVED** [review]
Made minor edits and placed `getClientStatus` in alphabetical order on the reference page. LGTM!

**Review by Montoya — APPROVED** [review]
LGTM

Another detail is when MetaMask is locked, dialogs do not work. We might want to update the documentation for dialogs to call this out.

**Comment by ziad-saab** [comment]
Thanks for the review. I'll add the bit on dialogs and re-request a review

**Review by Montoya — APPROVED** [review]

File: snaps/reference/snaps-api.md [hunk 1]
```diff
@@ -358,7 +362,44 @@ const addressKey1 = await deriveDogecoinAddress(1);
...
 
-## snap_getEntropy
+## `snap_getClientStatus`
+
+Gets the locked status of the Snaps client.
+
+It is useful to find out if MetaMask is locked in the following situations:
+
+- When running background operations that require MetaMask to be unlocked, for example [accessing encrypted state](#snap_managestate). If MetaMask is locked, the user gets a popup asking them to enter their password, which might be unexpected or confusing.
+- When [popping up a dialog](#snap_dialog). Dialogs do not work when MetaMask is locked.
```
Comment (alexandratran) [inline snaps/reference/snaps-api.md:372]:
```suggestion
It is useful to check if MetaMask is locked in the following situations:

- When running background operations that require MetaMask to be unlocked, for example, [accessing encrypted state](#snap_managestate). If MetaMask is locked, the user gets a popup asking them to enter their password, which might be unexpected or confusing.
- When [displaying a dialog](#snap_dialog). Dialogs do not work when MetaMask is locked.
```

File: snaps/reference/snaps-api.md [hunk 2]
```diff
@@ -9,12 +9,16 @@ Snaps can communicate with and modify the functionality of MetaMask using the [S
...
 
-## snap_dialog
+## `snap_dialog`
 
 Displays a dialog in the MetaMask UI.
 There are three types of dialogs with different parameters and return types: [alert](#alert-dialog),
 [confirmation](#confirmation-dialog), and [prompt](#prompt-dialog).
 
+:::caution
+Dialogs do not work when MetaMask is locked. To find out if MetaMask is locked, use the [`snap_getClientStatus` method](#snap_getclientstatus).
```
Comment (alexandratran) [inline snaps/reference/snaps-api.md:19]:
```suggestion
Dialogs do not work when MetaMask is locked. To check if MetaMask is locked, use [`snap_getClientStatus`](#snap_getclientstatus).
```

**Review by alexandratran — APPROVED** [review]

**Pull request merged**